### DISCUSSION TITLE
Git commit diffs

### DISCUSSION THREAD

**Discussion by rsdy** [GitoxideLabs/gitoxide#870] [Ideas]
We've made some progress on this with [Bloop](https://github.com/BloopAI/bloop/pull/566/files#diff-1697e771d622f3bb10da9f25d3a0cbf725c7d0eb4a70d67c5679b2a1261bd956R77), and this code seems to produce viable commits.

We're happy for you to internalize this tree traversal in `gix` if you think it's helpful, and I think I should have some time to make the patches with some guidance as well in the next month or so.

**Comment by Byron** [comment]
Thanks for sharing!

I only took a cursory look at the code and it wasn't immediately obvious how it's related to the diffing of trees, and I wonder why [`Tree::changes()`](https://github.com/Byron/gitoxide/blob/358500f0efaec7c67b307a6a1aa27ecad7502eb7/gix/tests/object/tree/diff.rs#L16) couldn't be used here. Maybe it wasn't discoverable enough? Please note that blob-diffs can also [conveniently be](https://github.com/Byron/gitoxide/blob/358500f0efaec7c67b307a6a1aa27ecad7502eb7/gix/tests/object/tree/diff.rs#L36) created with what's there.

Can you help me understand what the difference of your implementation is compared to what's already available?

**Reply by rsdy** [reply]
Sorry, the original message may not have been super clear. This code is to create git commits based on a list of diffs to files. I found that there's no way to do this currently withing `gix`, as the `changes()` only helps me create diffs based on existing commits.

I hope this clarifies the purpose of this.

**Reply by Byron** [reply]
Creating commits based on existing diffs, that sounds like `patch` or `git apply`? Is this what it is doing? Having general `apply` support would certainly be interesting, even though I'd be willing to wait it as there is too much on my plate already.
